### PR TITLE
[DEPRECATED] fix(coding-agent): normalize Windows NUL redirects to /dev/null

### LINKED ISSUE #4731: Windows NUL redirect creates spurious 'nul' files
## Summary

On Windows (e.g. PowerShell in Windows Terminal), pi spawns Git Bash internally to execute commands. AI-generated commands often contain `> /dev/null` or `2>/dev/null` redirects. In the MSYS2 runtime used by Git Bash, `nul` is not recognized as a device, so a plain file named `nul` is created in the working directory.

## Resolution

Fixed by [#4751](https://github.com/earendil-works/pi/pull/4751). `normalizeNulRedirects()` was added to rewrite Windows-style NUL redirects to `/dev/null` before spawning the shell.

### Behavior
- Only runs on `process.platform === 'win32'`
- Respects quoted strings (no replacement inside `'` or `"`)
- The token `nul` must be a standalone redirect target (followed by whitespace, end-of-string, or a shell control character) to avoid rewriting filenames like `nul.txt` or `nul-backup`
- File descriptors (`1`, `2`, `&`) must be adjacent to the operator (`1>/dev/null`) to be treated as descriptors; when separated by whitespace (`1 >/dev/null`) the number is preserved as a normal shell argument and only the redirect target is rewritten
- Escaped operators such as `\>` are literal arguments in Bash and are not normalized as redirects
- Backslashes inside single quotes are treated as literal characters (Bash semantics), so a command like `echo 'foo\' > nul` correctly exits the single-quote context and normalizes the redirect

### Affected variants
`>/dev/null`, `>>/dev/null`, `1>/dev/null`, `1>>/dev/null`, `2>/dev/null`, `2>>/dev/null`, `&>/dev/null`, `&>>/dev/null`

### PR DESCRIPTION
**Pull request by ayu-exorcist**
On Windows Terminal with PowerShell (and any other Windows terminal), pi spawns Git Bash to execute commands. AI-generated commands like `echo hello > nul` create a spurious file named `nul` because the MSYS2 runtime used by Git Bash does not treat `nul` as a null device.

This adds `normalizeNulRedirects()` which replaces NUL redirects outside of quoted strings with `/dev/null`, preserving the original operator (`>` or `>>`).

- Only runs on `process.platform === 'win32'`
- Respects quoted strings (no replacement inside `'` or `"`)
- Covers `> nul`, `>> nul`, `1> nul`, `2> nul`, `2>> nul`, `&> nul`, `&>> nul`

Fixes #4731

### CODE REVIEW

**Comment by ayu-exorcist** [comment]
Thanks for the review, Copilot. Both points have been addressed in a follow-up commit:

1. **Backslash-escaped quotes in `normalizeNulRedirects`**: Added an `isEscaped()` helper that counts consecutive backslashes before a quote character. Quotes now only toggle state when they are not escaped. This correctly handles `\"` (literal quote inside or outside double quotes) while preserving bash semantics where inside single quotes backslash has no special meaning.

2. **Integration test temp directory cleanup**: Replaced the `Date.now()` + `mkdirSync` approach with `mkdtempSync` and an `afterEach` hook that calls `rmSync(tempDir, { recursive: true, force: true })`, matching the pattern used by other tests in this package.

Additional test coverage was added for escaped-quote edge cases as well.

File: packages/coding-agent/src/core/tools/bash.ts [hunk 1]
```diff
@@ -56,6 +56,66 @@ export interface BashOperations {
...
+			i++;
+			continue;
+		}
+
+		if (char === '"' && !inSingleQuotes) {
+			inDoubleQuotes = !inDoubleQuotes;
+			result += char;
+			i++;
+			continue;
+		}
```
[the remark this answers is not in the document]
Comment (ayu-exorcist) [reply packages/coding-agent/src/core/tools/bash.ts:94]: Backslash-escaped quotes in normalizeNulRedirects: Added an isEscaped() helper that counts consecutive backslashes before a quote character. Quotes now only toggle state when they are not escaped. This correctly handles \" (literal quote inside or outside double quotes) while preserving bash semantics where inside single quotes backslash has no special meaning.

File: packages/coding-agent/test/bash-nul-redirect.test.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		const testDir = join(tmpdir(), `pi-nul-test-${Date.now()}`);
+		await import("node:fs").then((fs) => fs.mkdirSync(testDir, { recursive: true }));
+		const ops = createLocalBashOperations();
+
+		await ops.exec("echo hello > nul", testDir, {
+			onData: () => {},
+		});
+
+		expect(existsSync(join(testDir, "nul"))).toBe(false);
+	});
```
[the remark this answers is not in the document]
Comment (ayu-exorcist) [reply packages/coding-agent/test/bash-nul-redirect.test.ts:77]: Integration test temp directory cleanup: Replaced the Date.now() + mkdirSync approach with mkdtempSync and an afterEach hook that calls rmSync(tempDir, { recursive: true, force: true }), matching the pattern used by other tests in this package.
